server/routes/polls/polls.route.js

```diff
@@ -138,4 +139,102 @@ router.get(
   }
 );
 
+// @route GET /polls/:pollId
+// @desc To fetch a single poll using pollId
+// @params none
+// @access private
+router.get(
+  '/:pollId',
+  passport.authenticate('jwt', { session: true }),
+  async (req, res) => {
+    const poll = await Poll.findById(req.params.pollId);
+
+    if (!poll) {
+      return res.status(404).json({ message: 'Poll not found' });
+    }
+
+    return res.status(200).json(poll);
```

Comment:
Are all polls going to be public? This is something to think about in terms of should users only see polls that they are added to as a friends list? This would be something @tgnemecek considers though later when we have friends lists

Reply:
Personally, I think they should be private

If we make them private, I'll add this to `vote.validation.js`:
```JavaScript
  // Find and validate poll author
  const pollAuthor = await poll.populate('userId');
  if (!pollAuthor) {
    isValid = false;
    statusCode = 400;
    message = 'Poll author is no longer active on this platform';

    return { isValid, statusCode, message };
  }

  // Check if requesting user is a friend of the author of the poll
  if (!pollAuthor.friendIds.includes(req.user._id)) {
    isValid = false;
    statusCode = 401;
    message = 'You must follow poll author to vote on their poll';

    return { isValid, statusCode, message };
  }
```